src/serialize.h

```diff
@@ -272,7 +273,7 @@ void WriteCompactSize(Stream& os, const uint64_t nSize)
 }
 
 template<typename Stream>
-uint64_t ReadCompactSize(Stream& is)
+uint64_t ReadCompactSize(Stream& is, uint32_t max_size = MAX_DATA_SIZE)

```

Comment:
it supports 64-bit type, so max_size should be also the same type: uint64_t.
uint64_t ReadCompactSize(Stream& is, const uint64_t max_size = MAX_DATA_SIZE);
it's ok that MAX_DATA_SIZE is uint32_t: uint32_t -> uint64_t conversion is OK.